runtime/lua/vim/filetype.lua

```diff
@@ -2444,24 +2516,35 @@ local function expand_envvar_pattern(pat)
   return some_env_missing, expanded
 end
 
+--- Cache for whether current candidates match their parent patterns.
+--- Need to be manually reset in every new `M.match()` call.
+--- @type table<string,boolean>
+local parent_matches = {}
```

Comment:
Is there a reason to make this table scoped to the entire module and reset manually in `match()`, rather than just allocating the table in `match()` and passing it as an argument to `match_pattern_sorted`? If we can, we should try to keep the scope as narrow as possible.

Reply:
No reason apart from not having to make a six argument function. Will make it an argument.

Reply:
Addressed in 168d9e07fd855a2ef33bbe922b2295d9a9e7c3dd.